### ISSUE TITLE
swipl crashes on Mac OS 12.0.1

### ISSUE THREAD

**Issue by bdludwig** [SWI-Prolog/issues#107]
Hi,

since I updated my Mac to OS version 12.0.1 swipl crashes with a segmentation fault.
Now I tried to compile from the sources with the same issues coming up.

I attach the console output here:

bernd@Bernds-iMac build % cmake -G Ninja ..
-- Configuring SWI-Prolog-8.5.3
-- Using Macports packages from /opt/local
-- No junit.jar. Make sure junit.jar points to junit4.jar  Dropping JPL tests.
-- Could NOT find Qt5Widgets (missing: Qt5Widgets_DIR)
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/bernd/Software/swipl-devel/build
bernd@Bernds-iMac build % ninja
[142/417] Generating ../home/boot.prc
FAILED: home/boot.prc /Users/bernd/Software/swipl-devel/build/home/boot.prc 
cd /Users/bernd/Software/swipl-devel/build/src && /opt/local/bin/cmake -E remove -f /Users/bernd/Software/swipl-devel/build/home/boot.prc && /Users/bernd/Software/swipl-devel/build/src/swipl -q -O -o /Users/bernd/Software/swipl-devel/build/home/boot.prc -b /Users/bernd/Software/swipl-devel/build/home/boot/init.pl
/bin/sh: line 1: 28556 Segmentation fault: 11  /Users/bernd/Software/swipl-devel/build/src/swipl -q -O -o /Users/bernd/Software/swipl-devel/build/home/boot.prc -b /Users/bernd/Software/swipl-devel/build/home/boot/init.pl
ninja: build stopped: subcommand failed.

Any ideas why this happens and what could be a solution?

Thanks
Bernd

**Comment by JanWielemaker** [comment]
Yes.   The problem seems to be related to an interaction between MacOS locale handling and libtcmalloc.   I worked for my by using `cmake -DUSE_TCMALLOC=OFF ...`  Eventually our configuration should drop tcmalloc for this target (or this should be fixed in Apple's libc or the tcmalloc lib).
